annotation array initialization indentation issue

We upgraded Checkstyle from version 8.29 to 8.42, and we're seeing a strange violation around the indentation of annotation arrays.

Example:                                                                                                                                                                                                             
 
```
PS C:\abl\checkstyle-issue> type checkstyle-config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://checkstyle.sourceforge.net/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="Indentation"/>
    </module>
</module>

PS C:\abl\checkstyle-issue> type Story.java

/**
 * A class detailing fields common to all Story types.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY)
@JsonSubTypes(
    {
        @JsonSubTypes.Type(value = NonFictionStory.class, name = "NON_FICTION"),
        @JsonSubTypes.Type(value = MysteryStory.class, name = "MYSTERY"),
        @JsonSubTypes.Type(value = ScienceFictionStory.class, name = "SCIENCE_FICTION"),
        @JsonSubTypes.Type(value = HorrorStory.class, name = "HORROR"),
    }
)
public class Story {
}

PS C:\abl\checkstyle-issue> java -jar checkstyle-8.42-all.jar -c checkstyle-config.xml Story.java
Starting audit...
[ERROR] C:\abl\checkstyle-issue\Story.java:9:9: 'annotation array initialization' child has incorrect indentation level 8, expected level should be 4. [Indentation]
[ERROR] C:\abl\checkstyle-issue\Story.java:10:9: 'annotation array initialization' child has incorrect indentation level 8, expected level should be 4. [Indentation]
[ERROR] C:\abl\checkstyle-issue\Story.java:11:9: 'annotation array initialization' child has incorrect indentation level 8, expected level should be 4. [Indentation]
[ERROR] C:\abl\checkstyle-issue\Story.java:12:9: 'annotation array initialization' child has incorrect indentation level 8, expected level should be 4. [Indentation]
Audit done.
Checkstyle ends with 4 errors.
```
To clear the errors, I have to change the indentation of each @JsonSubTypes.Type in the array to be at the same indentation level as the curly brackets on lines 8 and 13. My understanding is using the default indentation value of 4, each array item should be indented 4 spaces from the curly brackets.

This behavior appears to have been introduced on version 8.40, if I downgrade Checkstyle to version 8.39, I don't have this problem.

I get a similar issue with 8.45 but this time Checkstyle reports a violation on the closing parenthesis if it is on a new line. This happens only in an array parameter of an annotation, not in the method body.

*AnnotationArray.java*
```java
public class AnnotationArray {

    @APIResponses({
        // OK
        @APIResponse(responseCode = "200", description = "Success"),
        // OK
        @APIResponse(
            responseCode = "404",
            description = "Not found"),
        // KO
        @APIResponse(
            responseCode = "500",
            description = "Internal error"
        ) // Error is here (line 14)
    })
    public void get() {
        // OK
        var foo = List.of(
            1,
            2,
            3
        );
        // OK
        var bar = new Integer[]{
            1,
            2,
            3
        };
    }

    private static @interface APIResponse {
        String responseCode();
        String description();
    }

    private static @interface APIResponses {
        APIResponse[] value() default {};
    }
}
```

*checkstyle.xml*
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN" "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <property name="charset" value="UTF-8"/>
    <property name="severity" value="error"/>

    <module name="SuppressWithPlainTextCommentFilter"/>
    <property name="fileExtensions" value="java, properties, xml"/>
    <module name="TreeWalker">
        <module name="Indentation">
            <property name="severity" value="error"/>
            <property name="basicOffset" value="4"/>
            <property name="braceAdjustment" value="0"/>
            <property name="caseIndent" value="4"/>
            <property name="throwsIndent" value="4"/>
            <property name="lineWrappingIndentation" value="4"/>
            <property name="arrayInitIndent" value="4"/>
        </module>
    </module>
</module>
```

```shell
java -jar checkstyle-8.45-all.jar -c checkstyle.xml AnnotationArray.java
```

```
Starting audit...
[ERROR] /home/hifi-filter.net/dussouille/Downloads/AnnotationArray.java:14:9: 'annotation array initialization' child has incorrect indentation level 8, expected level should be 12. [Indentation]
Audit done.
Checkstyle ends with 1 errors.
```

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/AnnotationArrayInitHandler.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/AnnotationArrayInitHandler.java
@@ -96,15 +96,24 @@ public class AnnotationArrayInitHandler extends BlockParentHandler {
         IndentLevel expectedIndent =
             new IndentLevel(getIndent(), getArrayInitIndentation(), getLineWrappingIndentation());
 
+        final DetailAST leftCurly = getLeftCurly();
         final int firstLine = getFirstLine(getListChild());
-        final int lcurlyPos = expandedTabsColumnNo(getLeftCurly());
+        final int lcurlyPos = expandedTabsColumnNo(leftCurly);
         final int firstChildPos =
             getNextFirstNonBlankOnLineAfter(firstLine, lcurlyPos);
+
+        // the code is written with old style where curlies are given their own line,
+        // the code block should be aligned to lcurly pos + lineWrappingIndent
+        if (firstChildPos == NOT_EXIST && lcurlyPos == getLineStart(leftCurly)) {
+            expectedIndent = IndentLevel.addAcceptable(expectedIndent,
+                        lcurlyPos + getLineWrappingIndentation());
+        }
 
         if (firstChildPos != NOT_EXIST) {
             expectedIndent = IndentLevel.addAcceptable(expectedIndent, firstChildPos, lcurlyPos
                     + getLineWrappingIndentation());
         }
+
         return expectedIndent;
     }
 

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java
@@ -602,6 +602,19 @@ public class IndentationCheckTest extends AbstractModuleTestSupport {
     }
 
     @Test
+    public void testAnnotationOddStyles() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(IndentationCheck.class);
+
+        checkConfig.addProperty("tabWidth", "8");
+
+        final String fileName = getPath("InputIndentationAnnotationArrayInitOldStyle.java");
+
+        final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
+
+        verifyWarns(checkConfig, fileName, expected);
+    }
+
+    @Test
     public void testZeroAnnotationArrayInit()
             throws Exception {
         final DefaultConfiguration checkConfig = createModuleConfig(IndentationCheck.class);
@@ -1225,8 +1238,8 @@ public class IndentationCheckTest extends AbstractModuleTestSupport {
             // following are tests for annotation array initialisation
             "120:13: " + getCheckMessage(MSG_CHILD_ERROR_MULTI, "annotation array initialization",
                 12, "16, 46, 48"),
-            "124:15: " + getCheckMessage(MSG_CHILD_ERROR, "annotation array initialization",
-                14, 12),
+            "124:15: " + getCheckMessage(MSG_CHILD_ERROR_MULTI, "annotation array initialization",
+                14, "12, 16"),
             "128:15: " + getCheckMessage(MSG_CHILD_ERROR_MULTI, "annotation array initialization",
                 14, "16, 28, 30"),
             "129:9: " + getCheckMessage(MSG_ERROR_MULTI, "annotation array initialization rcurly",
@@ -1312,8 +1325,8 @@ public class IndentationCheckTest extends AbstractModuleTestSupport {
             // following are tests for annotation array initialisation
             "120:13: " + getCheckMessage(MSG_CHILD_ERROR_MULTI, "annotation array initialization",
                 12, "16, 46, 48"),
-            "124:15: " + getCheckMessage(MSG_CHILD_ERROR, "annotation array initialization",
-                14, 12),
+            "124:15: " + getCheckMessage(MSG_CHILD_ERROR_MULTI, "annotation array initialization",
+                14, "12, 16"),
             "128:15: " + getCheckMessage(MSG_CHILD_ERROR_MULTI, "annotation array initialization",
                 14, "16, 28, 30"),
             "129:9: " + getCheckMessage(MSG_ERROR_MULTI, "annotation array initialization rcurly",

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationAnnotationArrayInitOldStyle.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationAnnotationArrayInitOldStyle.java
@@ -1,0 +1,26 @@
+/*                                                                          //indent:0 exp:0
+ * Indentation                                                              //indent:1 exp:1
+ * basicOffset = 4                                                          //indent:1 exp:1
+ * braceAdjustment = 0                                                      //indent:1 exp:1
+ * caseIndent = 4                                                           //indent:1 exp:1
+ * forceStrictCondition = false                                             //indent:1 exp:1
+ * lineWrappingIndentation = 4                                              //indent:1 exp:1
+ * tabWidth = 4                                                             //indent:1 exp:1
+ * throwsIndent = 4                                                         //indent:1 exp:1
+ */                                                                         //indent:1 exp:1
+
+package com.puppycrawl.tools.checkstyle.checks.indentation.indentation; //indent:0 exp:0
+
+@JsonSubTypes( //indent:0 exp:0
+    { //indent:4 exp:4
+        @Type(value="something") //indent:8 exp:8
+    } //indent:4 exp:4
+) //indent:0 exp:0
+
+@interface JsonSubTypes { //indent:0 exp:0
+    Type[] value() default {}; //indent:4 exp:4
+} //indent:0 exp:0
+
+@interface Type { String value() default "hello"; } //indent:0 exp:0
+
+public class InputIndentationAnnotationArrayInitOldStyle {} //indent:0 exp:0

